File: app/src/main/java/com/example/schooldiary/mainpage/bottomNavigation/NewsFragment.java
package com.example.schooldiary.mainpage.bottomNavigation;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.schooldiary.R;
import com.example.schooldiary.mainpage.model.News;
import com.example.schooldiary.mainpage.ui.FirestoreRecyclerAdapter;
import com.example.schooldiary.mainpage.ui.FirestoreRecyclerOptions;
import com.example.schooldiary.mainpage.ui.NewsHolder;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.Query;

public class NewsFragment extends Fragment {

    private RecyclerView newsRecyclerView;
    private FirestoreRecyclerAdapter adapter;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.fragment_news, container, false);
        newsRecyclerView = root.findViewById(R.id.news_recycler_view);
        return root;
    }

    @Override
    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {

        FirebaseFirestore firestore = FirebaseFirestore.getInstance();
        Query query = firestore.collection("news")
                .orderBy("serialNumber", Query.Direction.DESCENDING);

        FirestoreRecyclerOptions<News> options = new FirestoreRecyclerOptions.Builder<News>()
                .setQuery(query, News.class)
                .build();

        adapter = new FirestoreRecyclerAdapter<News, NewsHolder>(options) {
            @Override
            public void onBindViewHolder(@NonNull NewsHolder holder, int position, @NonNull News model) {
                holder.bindData(model);
            }

            @NonNull
            @Override
            public NewsHolder onCreateViewHolder(ViewGroup group, int i) {
                // Create a new instance of the ViewHolder, in this case we are using a custom
                // layout called R.layout.item_news for each item
                View view = LayoutInflater.from(group.getContext())
                        .inflate(R.layout.item_news, group, false);

                return new NewsHolder(view);
            }
        };

        newsRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        newsRecyclerView.setAdapter(adapter);
    }

    @Override
    public void onStart() {
        super.onStart();
        adapter.startListening();
    }

    @Override
    public void onStop() {
        super.onStop();
        adapter.stopListening();
    }
}

File: app/src/main/java/com/example/schooldiary/ui/main/news/NewsFragment.kt
package com.example.schooldiary.ui.main.news

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.navigation.Navigation
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.schooldiary.R
import com.example.schooldiary.databinding.FragmentNewsBinding
import com.example.schooldiary.model.dates.Dates
import com.example.schooldiary.model.news.News
import com.example.schooldiary.ui.main.news.NewsHolder.NewsClickListener
import com.example.schooldiary.viewmodel.news.NewsViewModel
import com.example.schooldiary.viewmodel.rings.CallScheduleViewModel
import com.firebase.ui.firestore.FirestoreRecyclerAdapter
import com.firebase.ui.firestore.FirestoreRecyclerOptions
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.Query

class NewsFragment : Fragment(), NewsClickListener {
    private var _binding: FragmentNewsBinding? = null
    private val binding get() = _binding!!
    private lateinit var newsFragmentAdapter: FirestoreRecyclerAdapter<News, NewsHolder>

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        _binding = FragmentNewsBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val viewModel: NewsViewModel by viewModels()
        val options = viewModel.getFirestoreRecyclerOptionsWithOrder<News>(
                collectionName = FIREBASE_NEWS_COLLECTION,
                order = FIREBASE_SERIAL_NUMBER_FIELD,
                queryDirection = Query.Direction.DESCENDING)
        newsFragmentAdapter = NewsFragmentAdapter(options, this)

        with(binding.recyclerView) {
            adapter = newsFragmentAdapter
            layoutManager = LinearLayoutManager(activity)
        }
    }

    override fun onStart() {
        super.onStart()
        newsFragmentAdapter.startListening()
    }

    override fun onStop() {
        super.onStop()
        newsFragmentAdapter.stopListening()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun openFragment(id: Int, photo: String, time: String, title: String, text: String) {
        val bundle = bundleOf(ARG_NEWS to News(photo, time, title, text))
        activity?.let {
            val navController = Navigation.findNavController(it, R.id.nav_host_fragment)
            navController.navigate(id, bundle)
        }
    }

    companion object {
        private const val FIREBASE_SERIAL_NUMBER_FIELD = "serialNumber"
        private const val FIREBASE_NEWS_COLLECTION = "news"
        private const val ARG_NEWS = "news"
    }
}